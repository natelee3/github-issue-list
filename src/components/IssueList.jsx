import React from 'react';
import Issue from './Issue';

class IssueList extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            issues: [],
        }
    }

    componentDidMount() {
        this._fetchIssues()
    }

    _fetchIssues = async () => {
        const url = 'https://api.github.com/repos/facebook/create-react-app/issues';
        const response = await fetch(url)
            .then(response => response.json())
        this.setState({
            issues: response
        })
        
    }

    render() {
        return (
            <div>
                <h2>facebook/<b>create-react-app</b></h2>
                <div class="container">
                    <table>
                        <tr class="border_bottom">
                            <td><b>916 Open </b>âœ“ 6,177 Closed </td>
                        </tr>
                        {this.state.issues.map((issue, index) => {
                                return (
                                    <Issue 
                                        key={index}
                                        url={issue.url}
                                        title={issue.title}
                                        number={issue.number}
                                        user={issue.user.login}
                                        body={issue.body}/>
                                )
                            })}
                    </table>
                </div>
            </div>
        )
    }


}

export default IssueList;