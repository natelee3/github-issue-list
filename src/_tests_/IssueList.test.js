
import React from 'react';
import IssueList from '../components/IssueList';
import renderer from 'react-test-renderer';

it('renders correctly', ()=> {
    const tree = renderer
      .create(<IssueList />)
      .toJSON();
    expect(tree).toMatchSnapshot()
})

