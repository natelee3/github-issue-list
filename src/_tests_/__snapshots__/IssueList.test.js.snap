// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2>
    facebook/
    <b>
      create-react-app
    </b>
  </h2>
  <div
    className="container"
  >
    <table>
      <thead />
      <tbody>
        <tr
          className="border_bottom"
        >
          <td>
            <b>
              916 Open 
            </b>
            âœ“ 6,177 Closed 
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
